gooddata-pandas/gooddata_pandas/result_convertor.py

```diff
@@ -0,0 +1,309 @@
+    """
+    Prepares header mapper function which is doing header structures translations into appropriate label used
+    in a dataframe
+    :param response: Response structure to gather dimension header details
+    :param dim: dimension id
+    :param label_overrides: label overrides
+    :return: Mapper function
+    """
+    dim_descriptor = response.dimensions[dim]
+    attribute_labels = label_overrides.get("labels", {})
+    measure_labels = label_overrides.get("metrics", {})
+
+    def _mapper(header_idx: int, header: Any) -> str:
+        label = ""
+        if header is None:
+            pass
+        elif "attributeHeader" in header:
+            if "labelValue" in header["attributeHeader"]:
+                label = header["attributeHeader"]["labelValue"]
+            elif "labelName" in header["attributeHeader"]:
+                attr_local_id = header["attributeHeader"]["localIdentifier"]
+                if attr_local_id in attribute_labels:
+                    label = attribute_labels[attr_local_id]["title"]
+                else:
+                    label = header["attributeHeader"]["labelName"]
+        elif "measureHeader" in header:
+            measure_idx = header["measureHeader"]["measureIndex"]
+            measure_descriptor = dim_descriptor["headers"][header_idx]["measureGroupHeaders"][measure_idx]
+
+            if measure_descriptor["localIdentifier"] in measure_labels:
+                label = measure_labels[measure_descriptor["localIdentifier"]]["title"]
+            elif "name" in measure_descriptor:
+                label = measure_descriptor["name"]
+            else:
+                label = measure_descriptor["localIdentifier"]
+        elif "totalHeader" in header:
+            label = header["totalHeader"]["function"]
+        return label
+
+    return _mapper
+
+
+def _headers_to_index(
+    dim_idx: int,
+    headers: Tuple[_DataHeaders, Optional[_DataHeaders]],
+    response: BareExecutionResponse,
+    label_overrides: LabelOverrides,
+) -> Optional[pandas.Index]:
+    if len(response.dimensions) <= dim_idx or not len(response.dimensions[dim_idx]["headers"]):
+        return None
+
+    mapper = _create_header_mapper(response=response, dim=dim_idx, label_overrides=label_overrides)
+
+    return pandas.MultiIndex.from_arrays(
+        [
+            tuple(mapper(header_idx, header) for header in header_group)
+            for header_idx, header_group in enumerate(cast(_DataHeaders, headers[dim_idx]))
+        ],
+        names=[mapper(0, dim_header) for dim_header in (response.dimensions[dim_idx]["headers"])],
+    )
+
+
+def _merge_grand_totals_into_data(extract: _DataWithHeaders) -> Union[_DataArray, List[_DataArray]]:
+    """
+    Merges grand totals into the extracted data. this function will mutate the extracted data, extending
+    the rows and columns with grand totals. Going with mutation here so as not to copy arrays around
+    """
+    data: Any = extract.data
```

Comment:
data type is Union[_DataArray, List[_DataArray]] right?

Reply:
It's actually `List[_DataArray]` - fixed